src/parse_util.rs

```diff
@@ -1402,6 +1413,39 @@ fn detect_errors_in_backgrounded_job(
     };
 
     let mut errored = false;
+
+    // Disallow background immediately before continuation conjunctions. For example:
+    // foo & && bar
+    // foo & || baz
+    if let Some(list) = job
+        .parent()
+        .unwrap()
+        .as_job_conjunction_continuation()
+        .and_then(|j| j.parent())
+        .and_then(|j| j.as_job_conjunction_continuation_list())
```

Comment:
I feel like this should probably not be done in `detect_errors_in_backgrounded_job()` but in the caller where we already check the parent type.
Else we might report the same error twice, no?

Reply:
You're right, this was probably a by-product of debugging attempts!